### PR TITLE
ACLs: Use fluffy in CI workflows

### LINKED ISSUE #8096: ACLs: Use fluffy in CI test environments
Use fluffy as external auth for CI tests

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #8096

### CODE REVIEW

File: esti/ops/docker-compose-dynamodb.yaml [hunk 1]
```diff
@@ -20,18 +31,51 @@ services:
...
-      - LAKEFS_DATABASE_KV_ENABLED=true
+      - LAKEFS_DATABASE_DYNAMODB_ENDPOINT=http://dynamodb:8000
       - LAKEFS_DATABASE_DYNAMODB_AWS_REGION=us-east-1
       - LAKEFS_DATABASE_DYNAMODB_AWS_ACCESS_KEY_ID=AKIAIO5FODNN7EXAMPLE
       - LAKEFS_DATABASE_DYNAMODB_AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K3MDENG/bPxRfiCYEXAMPLEKEY
-    entrypoint: ["/app/lakefs", "run"]
+    entrypoint: ["/app/wait-for", "fluffy:8000", "--", "/app/lakefs", "run"]
+
+  fluffy:
+    # TODO (niro): replace with 'latest'
```
Comment (arielshaqed) [inline esti/ops/docker-compose-dynamodb.yaml:43]: Please don't: that would make tests non-repeatable.  And that _increases_ the likelihood of breaking an enterprise customer.

If we must, we can add a matrix of versions to test.

**Review by arielshaqed — COMMENTED** [review]
I would approve the PR, except that I do not understand it.  Why are we doing this, rather than testing with the ACL server?

I would expect fluffy to test itself with a supported version of lakeFS, and for lakeFS tests to be self-contained.  Otherwise any code by an external OSS contributor will be untested _by definition_, at least until they open a PR.

File: esti/ops/docker-compose-dynamodb.yaml (hunk 1 above)
Comment (N-o-Z) [reply esti/ops/docker-compose-dynamodb.yaml:43]: What do you mean - that makes tests not repeatable?
We are not testing the fluffy server. We are testing lakeFS.
I'm not sure that this is the place to perform the this discussion in

File: pkg/api/controller.go [hunk 2]
```diff
@@ -4959,6 +4959,7 @@ func (c *Controller) Setup(w http.ResponseWriter, r *http.Request, body apigen.S
 	}
 
 	ctx := r.Context()
+	c.LogAction(ctx, "setup_lakefs", r, "", "", "")
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/api/controller.go:4962]: Removed

**Comment by N-o-Z** [comment]
> Setup

I think that from all possible options this might be the worst one.
The worst thing we can do is to start to pick and choose which tests are running in the esti jobs.
The tests are basically divided to 2 in this matter:
1. Uses auth operations - require auth server
2. Doesn't user auth operation - doesn't require auth server

I see no point in splitting execution just to run 2 distinct lists of tests.
I also don't understand how we can choose what is the right combination to run with auth server and what should we run without.
Should it be AWS / Azure / GCP, with/without external DB? I don't see any educated decision we can make here.

If this is the case - I prefer going with the original design decision to use the acl server for testing!

**Comment by itaiad200** [comment]
> > Setup
> 
> I think that from all possible options this might be the worst one. The worst thing we can do is to start to pick and choose which tests are running in the esti jobs. The tests are basically divided to 2 in this matter:
> 
> 1. Uses auth operations - require auth server
> 2. Doesn't user auth operation - doesn't require auth server
> 
> I see no point in splitting execution just to run 2 distinct lists of tests. I also don't understand how we can choose what is the right combination to run with auth server and what should we run without. Should it be AWS / Azure / GCP, with/without external DB? I don't see any educated decision we can make here.
> 
> If this is the case - I prefer going with the original design decision to use the acl server for testing!

If you test everything with external-auth, then you'll miss some important coverage for the simple-auth. For example, setup of lakeFS with simple-auth and the entire authorization system for all actions. For coverage reasons, I suggested at least 1 external-auth and 1 simple-auth. I think that duplicating all test suites for no-auth is extremely wasteful, that's why I suggested to split the existing tests. I agree there isn't an educated way to decide the proper combination for each. If you prefer you can duplicate one of the suites and have it tested with no-auth too.

**Comment by N-o-Z** [comment]
> > > Setup
> > 
> > 
> > I think that from all possible options this might be the worst one. The worst thing we can do is to start to pick and choose which tests are running in the esti jobs. The tests are basically divided to 2 in this matter:
> > 
> > 1. Uses auth operations - require auth server
> > 2. Doesn't user auth operation - doesn't require auth server
> > 
> > I see no point in splitting execution just to run 2 distinct lists of tests. I also don't understand how we can choose what is the right combination to run with auth server and what should we run without. Should it be AWS / Azure / GCP, with/without external DB? I don't see any educated decision we can make here.
> > If this is the case - I prefer going with the original design decision to use the acl server for testing!
> 
> If you test everything with external-auth, then you'll miss some important coverage for the simple-auth. For example, setup of lakeFS with simple-auth and the entire authorization system for all actions. For coverage reasons, I suggested at least 1 external-auth and 1 simple-auth. I think that duplicating all test suites for no-auth is extremely wasteful, that's why I suggested to split the existing tests. I agree there isn't an educated way to decide the proper combination for each. If you prefer you can duplicate one of the suites and have it tested with no-auth too.

We have one job dedicated to basic auth

File: esti/ops/docker-compose-dynamodb.yaml (hunk 1 above)
Comment (itaiad200) [reply esti/ops/docker-compose-dynamodb.yaml:43]: Replacing with `latest` means that tests may suddenly fail/pass due to Fluffy version bump. I agree that we should pin the version and that the `TODO` is redundant.

File: esti/ops/docker-compose-external-db.yaml
Comment (itaiad200) [inline esti/ops/docker-compose-external-db.yaml:1]: What's the difference between this file and `docker-compose-dynamodb.yaml`? It seems to me like they are used for the same configuration. If that's the case, let's consolidate them?

**Review by itaiad200 — APPROVED** [review]
Thank you! One comment on the option to consolidate docker-compose files - I think that there are 2 that repeat themselves. Sorry about the back and forth on basic-auth - It was added before this PR and I missed that 🥲

**Comment by arielshaqed** [comment]
Given that we are merging this, please open an issue that this introduces a circular testing dependency and also makes lakeFS untestable without a closed-source component.  I would like us to be able to revisit this.

File: esti/ops/docker-compose-dynamodb.yaml (hunk 1 above)
Comment (N-o-Z) [reply esti/ops/docker-compose-dynamodb.yaml:43]: Done

File: esti/ops/docker-compose-external-db.yaml
Comment (N-o-Z) [reply esti/ops/docker-compose-external-db.yaml:1]: The difference is in the lakeFS configuration. We might be able to consolidate them but I don't think it's worth spending time on it now.
I'll open an issue for composition of yaml files

Comment (N-o-Z) [reply esti/ops/docker-compose-external-db.yaml:1]: https://github.com/treeverse/lakeFS/issues/8122

**Comment by N-o-Z** [comment]
> Given that we are merging this, please open an issue that this introduces a circular testing dependency and also makes lakeFS untestable without a closed-source component. I would like us to be able to revisit this.

https://github.com/treeverse/lakeFS/issues/8123

**Pull request merged**